examples/example_dotnet/WinSparkleDemoForm.cs

```diff
@@ -0,0 +1,155 @@
+        {
+            _sparkleNet.CheckForUpdate(true, true);
+        }
+
+        private void btn_close_Click(object sender, EventArgs e)
+        {
+            this.Close();
+        }
+
+        private bool CanShutDownCallback()
+        {
+            try
+            {
+                var result = MessageBox.Show(@"Here you can simulate if the application can be closed or not to proceed the update", @"CanShutDownCallback",
+                    MessageBoxButtons.YesNo, MessageBoxIcon.Question);
+                return result == DialogResult.Yes;
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show(ex.Message, @"Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
+                return false;
+            }
+        }
+
+        private void DidFindAnUpdateCallback()
+        {
+            try
+            {
+                MessageBox.Show(@"Callback called if an update has been found.", @"DidFindUpdateCallback",
+                    MessageBoxButtons.OK, MessageBoxIcon.Information);
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show(ex.Message, @"Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
+            }
+        }
+
+        private void DidNotFindAnUpdateCallback()
+        {
+            try
+            {
+                MessageBox.Show(@"Callback called if an update hasn't been found.", @"DidNotFindUpdateCallback",
+                    MessageBoxButtons.OK, MessageBoxIcon.Information);
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show(ex.Message, @"Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
+            }
+        }
+
+        private void UpdateCancelledCallback()
+        {
+            try
+            {
+                MessageBox.Show(@"Callback called if an update has been cancelled/ignored by user.", @"UpdateCancelledCallback",
+                    MessageBoxButtons.OK, MessageBoxIcon.Information);
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show(ex.Message, @"Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
+            }
+        }
+
+        private void ErrorCallback()
+        {
+            try
+            {
+                MessageBox.Show(@"Callback called if WinSparkle get an error.", @"ErrorCallback",
+                    MessageBoxButtons.OK, MessageBoxIcon.Information);
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show(ex.Message, @"Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
+            }
+        }
+
+        private void ShutDownRequestCallback()
+        {
+            try
+            {
+                if (System.Windows.Forms.Application.MessageLoop)
+                {
+                    System.Windows.Forms.Application.Exit();
+                }
+                else
+                {
+                    System.Environment.Exit(1);
```

Comment:
As @Phrohdoh says. Additionally, why do you have this here in the first place? Surely as the developer of the example, you *know* whether it is running a message loop or not, don’t you?